src/datasources/asset/types.ts

```diff
@@ -0,0 +1,140 @@
+import { DataQuery } from '@grafana/schema'
+
+export interface AssetQuery extends DataQuery {
+  type: AssetQueryType,
+  workspace: string,
+  minionIds: string[]
+}
+
+export enum AssetQueryType {
+  Metadata = "Metadata",
+}
+
+export enum EntityType {
+  Asset = "Asset",
+  System = "System"
+}
+
+export interface AssetsResponse {
+  assets: AssetModel[],
+  totalCount: number
+}
+
+export interface AssetModel {
+  modelName: string,
+  modelNumber: number,
+  serialNumber: string,
+  vendorName: string,
+  vendorNumber: number,
+  busType: 'BUILT_IN_SYSTEM' | 'PCI_PXI' | 'USB' | 'GPIB' | 'VXI' | 'SERIAL' | 'TCP_IP' | 'CRIO' | 'SCXI' | 'CDAQ' | 'SWITCH_BLOCK' | 'SCC' | 'FIRE_WIRE' | 'ACCESSORY' | 'CAN' | 'SWITCH_BLOCK_DEVICE' | 'SLSC',
```

Comment:
What happens if we add a new bus type in the service? I guess it involves updating the plugin. My question is whether we define this as a generic string, so the client is more flexible for modifications. I think this is possible if the code is not doing particular checks with these values.

This applies for other enums as well.

Reply:
can we just add string type in the end?

Reply:
Yes, I agree

Reply:
all enums that will not be used in code(if-else statments or etc) are replaced with strings